python/lsst/daf/butler/cli/butler.py

```diff
@@ -255,14 +277,14 @@ def _cmdNameToFuncName(cls, commandName: str) -> str:
 
     @staticmethod
     def _mergeCommandLists(
-        a: defaultdict[str, list[str]], b: defaultdict[str, list[str]]
-    ) -> defaultdict[str, list[str]]:
+        a: defaultdict[str, list[str | PluginCommand]], b: defaultdict[str, list[str | PluginCommand]]
+    ) -> defaultdict[str, list[str | PluginCommand]]:
         """Combine two dicts whose keys are strings (command name) and values
         are list of string (the package(s) that provide the named command).
 
         Parameters
         ----------
-        a : `defaultdict` [`str`, `list` [`str`]]
+        a : `defaultdict` [`str`, `list` [`str` | `PluginCommand` ]]
```

Comment:
It's specifically designed to complain if two packages define a subcommand of the same name so the `list` here is deliberate. There is a later check that makes sure that there is only one entry in the list.

Reply:
But if two packages are the same package? I'm thinking about the case when a package has both an entry point and envar?

Reply:
1. If people don't need to set the envvar if they are using pip and they can't use entry points if they are using EUPS so I don't think it's a problem as such (it will mostly be for people who have been using pip and setting the envvar and it's better if they stop doing that anyhow rather than continuing to think they need the envvar.
2. Currently you get two strings in the comparison: lsst.pipe.base.cli.cmd and lsst.pipe.base.cli.get_cli_subcommands. If you know that the plugin is going to be in `lsst.pipe.base.cli` then you can work it out but in the general case I think you would need to load all the commands and do `is` check and then warn if they are the same -- I'm not sure it's worth it given point 1. We don't want pip users to set the env vars.

Reply:
OK, I don't want to overcomplicate it, fine to leave it as it is 👍